client/src/main/java/io/pravega/client/stream/notifications/notifier/AbstractNotifier.java

```diff
@@ -18,7 +18,7 @@
 
 /**
  * AbstractNotifier which is used by all types of Notifiers.
- * @param <T> The type of notification being listened to.
+ * @param <T> notification template
```

Comment:
I'd not say template here. I would either stay at `notification`  or say `notification subtype`. But, it might be more interesting to say that this is the type of notifications that this notifier is issuing, or some rephrase of it. Otherwise we are just stating the obvious, while the goal is to help the developer understand what this type is about.